### PR TITLE
[fix][cdc-base] fix cdc bug

### PR DESCRIPTION
**Pull request by fuyun2024**
issues : [issues-1122](https://github.com/ververica/flink-cdc-connectors/issues/1122)

### CODE REVIEW

File: flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceFetchTaskContext.java [hunk 1]
```diff
@@ -83,4 +84,6 @@ public SchemaNameAdjuster getSchemaNameAdjuster() {
     public abstract ChangeEventQueue<DataChangeEvent> getQueue();
 
     public abstract Offset getStreamOffset(SourceRecord sourceRecord);
+
+    public abstract RelationalTableFilters getRelationalTableFilters();
```
Comment (ruanhang1993) [inline flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceFetchTaskContext.java:88]: It seems that this new method is not relational to this bug.

File: flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceStreamFetcher.java [hunk 2]
```diff
@@ -79,7 +79,9 @@ public JdbcSourceStreamFetcher(JdbcSourceFetchTaskContext taskContext, int subTa
     public void submitTask(FetchTask<SourceSplitBase> fetchTask) {
         this.streamFetchTask = fetchTask;
         this.currentStreamSplit = fetchTask.getSplit().asStreamSplit();
+        configureFilter();
```
Comment (ruanhang1993) [inline flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceStreamFetcher.java:82]: I think invoking this method is enough to fix this bug. Why do we need other modifies?

Comment (molsionmo) [reply flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceStreamFetcher.java:82]: I just invoke this method and have solved that problem by retesting it

**Review by molsionmo — COMMENTED** [review]
I think the change of [JdbcSourceStreamFetcher.java] is enough for the fix

File: flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceScanFetcher.java [hunk 3]
```diff
@@ -113,6 +113,7 @@ public Iterator<SourceRecord> pollSplitRecords() throws InterruptedException {
             boolean reachBinlogEnd = false;
             final List<SourceRecord> sourceRecords = new ArrayList<>();
             while (!reachBinlogEnd) {
+                checkReadException();
```
Comment (molsionmo) [inline flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/JdbcSourceScanFetcher.java:116]: I think it solves the problem #1098. It is better that add a test for JdbcSourceScanFetcher.java like com.ververica.cdc.connectors.mysql.debezium.reader.SnapshotSplitReaderTest#testThrowRuntimeExceptionInSnapshotScan

File: flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/experimental/fetch/MySqlSourceFetchTaskContext.java [hunk 4]
```diff
@@ -221,6 +222,11 @@ public Offset getStreamOffset(SourceRecord sourceRecord) {
         return MySqlUtils.getBinlogPosition(sourceRecord);
     }
 
+    @Override
+    public RelationalTableFilters getRelationalTableFilters() {
+        return getDbzConnectorConfig().getTableFilters();
+    }
```
Comment (molsionmo) [inline flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/experimental/fetch/MySqlSourceFetchTaskContext.java:228]: It seems that this new method is not relational to this bug.

**Comment by molsionmo** [comment]
fix: #1122

**Review by molsionmo — APPROVED** [review]

**Review by leonardBang — APPROVED** [review]
Thanks @fuyun2024 and @molsionmo for the contribution, sorry for the late reply, LGTM.

**Pull request merged**